spacy/cli/_util.py

```diff
@@ -331,44 +331,49 @@ def import_code(code_path: Optional[Union[Path, str]]) -> None:
             msg.fail(f"Couldn't load Python code: {code_path}", e, exits=1)
 
 
-def upload_file(src: Path, dest: Union[str, "Pathy"]) -> None:
+def upload_file(src: Path, dest: Union[str, Path, "CloudPath"]) -> None:
     """Upload a file.
 
     src (Path): The source path.
     url (str): The destination URL to upload to.
     """
-    import smart_open
+    dest = ensure_pathy(dest)
 
-    dest = str(dest)
-    with smart_open.open(dest, mode="wb") as output_file:
+    # Create parent directories for local paths
+    if isinstance(dest, Path):
+        if not dest.parent.exists():
+            dest.parent.mkdir(parents=True)
+
+    with dest.open(mode="wb") as output_file:
         with src.open(mode="rb") as input_file:
             output_file.write(input_file.read())
 
 
-def download_file(src: Union[str, "Pathy"], dest: Path, *, force: bool = False) -> None:
+def download_file(
+    src: Union[str, Path, "CloudPath"], dest: Path, *, force: bool = False
+) -> None:
     """Download a file using smart_open.
 
     url (str): The URL of the file.
     dest (Path): The destination path.
     force (bool): Whether to force download even if file exists.
         If False, the download will be skipped.
     """
-    import smart_open
+    src = ensure_pathy(src)
 
     if dest.exists() and not force:
         return None
-    src = str(src)
-    with smart_open.open(src, mode="rb", ignore_ext=True) as input_file:
+    with src.open(mode="rb") as input_file:
```

Comment:
This would need to be reverted with `smart_open` added as a separate dependency because it's also used in `spacy project assets` and not just remote storage.